scripts/linksys-header.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/python3
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# Copyright (C) 2024 OpenWrt.org
+#
+# ./linksys-header.py <ImageFileIn> <ImageFileOut> <ModelID>
+#
+# This script adds an image header for Linksys switches of the LGS3xxC/LGS3xxMPC
+# series. It resembles a U-Boot legacy format image header, all data in network
+# byte order (aka natural aka bigendian). Possible values for ModelID are
+# 
+# "BKS-RTL83xx" : smaller models with RTL83xx SOC
+# "BKS-RTL93xx" : larger models with RTL93xx SOC
+
+import argparse
+import os
+import zlib
+import array
+
+MAGIC83 = 0x07800001
+MAGIC93 = 0x07600001
+VERSION1 = 1
+VERSION2 = 1
+VERSION3 = 2
+VERSION4 = 2
+COMPANY = "belkin"
+MODULE = "IMG"
+
+def xcrc32(buf):
+    return (0xffffffff - zlib.crc32(buf, 0xffffffff)).to_bytes(4, 'big')
+
+def encode_model(model):
+    map = " 0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz-"
+    code = bytearray()
+
+    code.append(map.index(model[:1]))
+    model = model[1:]
+    model = model + " " * (3 - len(model) % 4)
+
+    while model != "":
+        b1 = map.index(model[0:1])
+        b2 = map.index(model[1:2])
+        b3 = map.index(model[2:3])
+        b4 = map.index(model[3:4])
+        model = model[4:]
+
+        code.append(b1 << 2 | b2 >> 4)
+        code.append((b2 & 0xf) << 4 | b3 >> 2)
+        code.append((b3 & 0x3) << 6 | b4)
+
+    return code
+
+def create_header(buf, model):
+    head = bytearray(32)
+
+    if model == "BKS-RTL83xx":
+        head[0:4] = MAGIC83.to_bytes(4, 'big')
+    else:
+        head[0:4] = MAGIC93.to_bytes(4, 'big')
```

Comment:
If more models are expected to be added, it would be more flexible to also provide this header magic via the device recipe in the Makefile

Reply:
Will leave as is. For now only these two types where identified